cucumber/authenticators_gcp/features/step_definitions/authn_gcp_steps.rb

```diff
@@ -0,0 +1,23 @@
+
+Given(/^I set a GCP annotation with name: "([^"]*)" and value: "([^"]*)" to host: "([^"]*)"$/) do |hostname, annotation_name, annotation_value|
+    i_have_a_resource "host", hostname
+
+    if AuthenticatorHelpers.method_defined? annotation_value
+      set_annotation_to_resource(annotation_name,
+                                 Class.new.extend(AuthenticatorHelpers).send(annotation_value.to_s))
+    else
+      set_annotation_to_resource(annotation_name, annotation_value)
+    end
```

Comment:
what's going on here? when will this method be defined and when not?

Reply:
If you want a static value you pass a string if you want cal a method in the helper class you pass its name

Reply:
that's not what i asked. i wanted to know when will we go into each `if` clause. can you show me 2 usages of this function? it is not clear from the code

Reply:
currently there is only one use of that method:
`annotation with name: "authn-gcp/service-account-id" and value: "gcp_service_account_id"`

Reply:
implemented using constants